templates/supplier_app/invoices_form.html

```diff
@@ -15,142 +15,169 @@ <h5 class="alert alert-danger">{{ err }}</h5>
+                    <label class="custom-file-label" id='id-custom-file-label1' for="inputGroupFile01">Choose file</label>
+                  </div>
                 </div>
-                <div class="col-md-12 text-center">
-                    <button id="submit-btn" type="submit" class="btn btn-primary">
-                        {% trans "Submit" %}
-                    </button>
-                    <a class='btn btn-danger' href="{% url 'supplier-invoice-list' taxpayer_id=taxpayer_id %}">
-                        {% trans "Cancel" %}
-                    </a>
-                </div>
-                {% csrf_token %}
-            </form>
+          </div>
+          {% if form.invoice_file.errors %}
+            {% for error in form.invoice_file.errors %}
+              <div class="alert alert-danger alert-dismissible fade show mb-0" role="alert">
+                {{ error }}
+                <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+                  <span aria-hidden="true">&times;</span>
+                </button>
+              </div>
+            {% endfor %}
+          {% endif %}
+          <div class="row ml-0 mr-0 mt-3">
+            <div class="input-group">
+              <div class="input-group-prepend">
+                <span class="input-group-text" id="inputGroupFileAddon01">PO File</span>
+              </div>
+              <div class="custom-file">
+                {{ form.po_file }}
+                <label class="custom-file-label" id='id-custom-file-label2' for="inputGroupFile01">Choose file</label>
+              </div>
+            </div>
+          </div>
+          {% if form.po_file.errors %}
+            {% for error in form.po_file.errors %}
+              <div class="alert alert-danger alert-dismissible fade show mt-3 mb-0" role="alert">
+                {{ error }}
+                <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+                  <span aria-hidden="true">&times;</span>
+                </button>
+              </div>
+            {% endfor %}
+          {% endif %}
         </div>
-
+        <div class="col-md-12 text-right mt-3">
+          <button id="submitBtn" type="submit" class="btn btn-primary">
+            Submit
+          </button>
+          <a class='btn btn-danger' href="{% url 'supplier-invoice-list' taxpayer_id=taxpayer_id %}">Cancel</a>
+        </div>
+        {% csrf_token %}
+      </form>
     </div>
+
+  </div>
 </div>
 
 </script>
 <script type="text/javascript">
-$(document).ready(function () {
-    $("#invoice-form").submit(function () {
-        $("#submit-btn").attr("disabled", true);
-        return true;
+  $(function() { {
+    $("#invoiceForm").submit(function() {
+      $("#submitBtn").attr("disabled", true);
+      return true;
     });
-});
+  });
+
+  $('#inputGroupFile01').change(function(e) {
+    $(this).next('#id-custom-file-label1').html(e.target.files[0].name);
```

Comment:
Are you sure that the target always will have `files`?

Reply:
If there'sn no file, it will be validated in the back end and a new page will be served with alerts about "No File", "File Size Greater than ..." and "File isn't a PDF"